pkg/cloud/services/compute/instance.go

```diff
@@ -376,38 +409,73 @@ func getServerNetworks(networkClient *gophercloud.ServiceClient, networkParams [
 	return nets, nil
 }
 
-func getOrCreatePort(is *Service, eventObject runtime.Object, clusterName string, portName string, net infrav1.Network, securityGroups *[]string) (*ports.Port, error) {
+func getOrCreatePort(is *Service, eventObject runtime.Object, clusterName string, portName string, net infrav1.Network, instanceSecurityGroups *[]string) (*ports.Port, error) {
 	allPages, err := ports.List(is.networkClient, ports.ListOpts{
 		Name:      portName,
 		NetworkID: net.ID,
 	}).AllPages()
 	if err != nil {
 		return nil, fmt.Errorf("searching for existing port for server: %v", err)
 	}
-	portList, err := ports.ExtractPorts(allPages)
+	existingPorts, err := ports.ExtractPorts(allPages)
 	if err != nil {
 		return nil, fmt.Errorf("searching for existing port for server: %v", err)
 	}
 
-	if len(portList) != 0 {
-		return &portList[0], nil
+	if len(existingPorts) == 1 {
+		return &existingPorts[0], nil
 	}
 
-	portCreateOpts := ports.CreateOpts{
+	if len(existingPorts) > 1 {
+		return nil, fmt.Errorf("multiple ports found with name \"%s\"", portName)
+	}
+
+	// no port found, so create the port
+	description := net.PortOpts.Description
+	if description == "" {
+		description = fmt.Sprintf("Created by cluster-api-provider-openstack cluster %s", clusterName)
```

Comment:
Should be adjusted once: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/870/files is merged (which should be soon)

Reply:
I've updated to use this new function to get the default description (if none is provided in the port options).